<launch>

<node pkg="sr_slam" type="test_vo_rs" name="test_vo_rs" output="screen">
  <!-- sift parameters -->
  <param name="sift_num_features"                   value="0" /> <!-- number of returned sift features -->
  <param name="sift_octave_layers"                  value="3" /> <!-- number octave layers -->
  <param name="sift_contrast_threshold"             value="0.08" /> <!-- 0.04 initial threshold for keypint detection -->
  <param name="siftgpu_contrast_threshold"          value="0.0018" /> <!-- 0.0018 0.0008 initial DOG threshold for siftgpu -->
  <!-- param name="siftgpu_edge_threshold"              value="10." /--> <!-- edge threshold for siftgpu  -->

  <!-- Visual Features, to activate GPU-based features see CMakeLists.txt -->
  <param name="config/feature_detector_type"         value="SIFTGPU"/><!-- If SIFTGPU is enabled in CMakeLists.txt, use SURF here -->
  <param name="config/feature_extractor_type"        value="SIFTGPU"/><!-- If SIFTGPU is enabled in CMakeLists.txt, use SURF here -->
  <param name="config/min_keypoints"                 value="20"/><!-- Extract no less than this many ... -->
  <param name="config/min_matches"                   value="12"/> <!-- Minimal number of matched features before RANSAC 29? -->
  <param name="config/max_keypoints"                 value="10000"/> <!-- maximum number of key points -->
  <param name="config/nn_distance_ratio"             value="0.95"/> <!-- 0.5/ 0.7 for SIFT, 0.95 for SIFTGPU Feature correspondence is valid if distance to nearest neighbour is smaller than this parameter times the distance to the 2nd neighbour. This needs to be 0.9-1.0 for SIFTGPU w/ FLANN, since SIFTGPU Features are normalized-->

  <!-- max rotation and translation for a VO match between two frames -->
  <param name="config/max_rotation_degree"          value="90"/> 
  <param name="config/max_translation_meter"        value="1.5"/>

  <param name="config/max_dist_for_inliers"          value="0.05"/> <!-- 0.05 distance for matches to be considered inliers by ransac-->
  <param name="config/ransac_iterations"             value="10000" /> <!-- 200 -->
  <param name="config/b_ransac_fixed_iter"           value="false"/> <!-- weather to use fixed   -->

  <!-- vro strategy -->
  <param name="config/vro_strategy"                 value="vro_my" /> <!-- " VRO strategy: VRO_ori, VRO_my, VRO_plane" -->

  <!-- vro parameters -->
  <param name="rs_data_file_dir"        value="/media/work/work/data/realsense/10-23-2015_realsense_R200_etas_2nd_floor/captured_data"/>
  <!-- param name="rs_data_file_dir"        value="/media/work/work/data/realsense/captured_data_newspaper_c"/-->
  <!-- param name="rs_data_file_dir"        value="/media/work/work/data/realsense/captured_data_EIT_5th_floor"/-->

  <param name="rs_data_prefix"                value="f"/>
  <param name="rs_data_suffix"                value="dat"/> <!-- dat -->
  <param name="rs_start_frame"                value="1"/> <!-- notice: for matching timestamp in ground_truth, start should be 1 -->
  <param name="rs_data_strip"                 value="5"/> <!-- strip between two rs frames to be implemented -->
  <param name="rs_end_frame"                  value="360"/> 

  <param name="vo_rs_model"                 value="sequence_match"/> <!-- single_match, sequence_match -->
  <param name="vro_pack_number"                     value="500"/>
  <param name="vro_src_id"                          value="10" />
  <param name="vro_tar_id"                          value="15" />
  <param name="vro_single_times"                    value="1" /> 
  <param name="vro_display_matches"                 value="true" /> <!-- weather to display the matched points -->
  <!-- param name="vro_match_result_log"     value="/home/davidz/work/data/SwissRanger4000/vro_results/bdat/angle_test/m_result_9.log"/-->
  <!-- param name="vro_statistic_result_log" value="/home/davidz/work/data/SwissRanger4000/vro_results/bdat/angle_test/s_result_9.log"/-->
  <param name="vro_match_result_log"     value="/home/davidz/work/data/SwissRanger4000/vro_results/bdat/angle_test/pitch_v2/m_result_p15.log"/>
  <param name="vro_statistic_result_log" value="/home/davidz/work/data/SwissRanger4000/vro_results/bdat/angle_test/pitch_v2/s_result_p15.log"/>

 </node>
</launch>
